docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,571 @@
+        'covid-csv': {
+            'url': args.index_data_url,
+            'filename': os.path.join(args.workdir, 'dataset.csv'),
+        }
+    }
+    # download the data
+    download_data(targets, args.download_proxy, task_name='download covid-dataset')
+    flow = (
+        Flow(cors=True)
+            .add(name='MyTransformer', uses=MyTransformer)
+            .add(name='MyIndexer', uses=MyIndexer)
+            .plot('test.svg')
+    )
+    with flow, open(targets['covid-csv']['filename']) as fp:
+        flow.index(from_csv(fp, field_resolver={'question': 'text'}))
+    
+        # switches the serving protocol to HTTP at runtime
+        flow.protocol = 'http'
+        flow.port_expose = args.port_expose
+        url_html_path = 'file://' + os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.realpath(__file__)), 'static/index.html'
+            )
+        )
+        try:
+            webbrowser.open(url_html_path, new=2)
+        except:
+            pass  # intentional pass, browser support isn't cross-platform
+        finally:
+            default_logger.success(
+                f'You should see a demo page opened in your browser, '
+                f'if not, you may open {url_html_path} manually'
+            )
+        if not args.unblock_query_flow:
+            flow.block()
+```
+
+```{admonition} See Also
+:class: seealso
+For more information on what the Flow is doing, and how to serve the flow with `f.block()` and configure the protocol, 
+check {ref}`the Flow fundamentals section <flow-cookbook>`.
+```
+
+```{admonition} Important
+:class: important
+Since we want to call our Flow from the browser, it's important to enable 
+[Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) with `Flow(cors=True)`
+```
+
+
+
+Ok, so it seems that we have plenty of work done already. If you run this you will see a new tab open in your browser, and there you will have a text box ready for you to input some text. However, if you try to enter anything you won't get any results. This is because we are using dummy Executors. Our `MyTransformer` and `MyIndexer` aren't actually doing anything. So far they only print a line when they are called. So we need real Executors.
+
+This has been plenty of new information you've learned so far, so we won't go deep into Executors today. Instead you can copy-paste the ones we are using for [this example](https://github.com/jina-ai/jina/blob/master/jina/helloworld/chatbot/my_executors.py), save that `my_executors.py` file in the same directory where the rest of your code is. The important part to understand is that all Executors' behavior is defined in `my_executors.py`
+
+To try the Executors from the Github repo, just add this before the `download_data` function:
+
+``` python
+if __name__ == '__main__':
+    from my_executors import MyTransformer, MyIndexer
+else:
+    from .my_executors import MyTransformer, MyIndexer
+```
```

Comment:
Why not create actual executors here? Or use ones form the hub? Having to copy-paste from some link does not seem user friendly

Reply:
No executors are created in the tutorial with a brief explanation.
I  chose to split the implementation of the executors because it's easier to understand and also I didn't want to embed very large code snippets (don't have a way to collapse code blocks atm)